### ISSUE TITLE
Trying to validate quality of generated samples

### ISSUE THREAD

**Issue by m-mandel** [freds0/free-svc#17]
Hi,

Thank you for your work.

I downloaded the pretrained ckpt from huggingface.
I have applied your work on some samples from librispeech, and got some generated samples which I am including here:

[gen-samples-spin-language-emb.tar.gz](https://github.com/user-attachments/files/22212730/gen-samples-spin-language-emb.tar.gz)

Is this what I am expected to get?
Am I running the script correctly? 

This was the command and output log:

> 
> python free-svc/scripts/inference_sample.py \
>     --hpfile free-svc/configs/config-online-spin-language-emb.yaml \
>     --ptfile free-svc/G_00014_0225000.pth \
>     --source-path <data_dir_path>/2_src.wav \
>     --target-path <data_dir_path>/2_ref_tgt1.wav \
>     --src-lang english
>     
> WARNING:root:TTS is not installed, COQUI speaker encoder will not be available
> Downloading: "https://github.com/snakers4/silero-vad/zipball/master" to /root/.cache/torch/hub/master.zip
> {'defaults': ['common'], 'train': {'batch_size': 128, 'betas': [0.8, 0.99], 'c_kl': 1.0, 'c_mel': 45, 'distributed': False, 'use_multiprocessing': False, 'epochs': 20, 'eps': '1e-9', 'fp16_run': False, 'init_lr_ratio': 1, 'raise_error': False, 'learning_rate': '2e-4', 'log_interval': 10, 'log_level': '${log_level}', 'lr_decay': 0.98, 'max_speclen': 128, 'port': 8005, 'resume_training': False, 'seed': 1234, 'segment_size': 8960, 'use_sr': False, 'valid_epoch_interval': 1, 'valid_steps_interval': 1000, 'save_epoch_interval': 10, 'save_steps_interval': 1000, 'warmup_epochs': 0, 'weighted_batch_speaker_sampling': 0.5, 'weighted_batch_lang_sampling': 0.5}, 'data': {'dataset_dir': '/raid/lucasgris/free-svc/data', 'filter_length': 1280, 'hop_length': 320, 'max_wav_value': 32768.0, 'mel_fmax': None, 'mel_fmin': 0.0, 'n_mel_channels': 80, 'num_workers': 64, 'pitch_predictor': 'rmvpe', 'pitch_features_dir': '${data.dataset_dir}/pitch_features/', 'sampling_rate': 24000, 'spectrogram_dir': None, 'sr_min_max': [68, 92], 'content_feature_dir': None, 'training_files': 'data/train.csv', 'validation_files': 'data/valid.csv', 'win_length': 1280, 'use_lang_emb': True, 'num_langs': 11, 'lang_dim': 192, 'lang2id': {'chinese': 0, 'dutch': 1, 'english': 2, 'french': 3, 'german': 4, 'italian': 5, 'japanese': 6, 'other': 7, 'polish': 8, 'portuguese': 9, 'spanish': 10}, 'use_spk_emb': False, 'spk_embeddings_dir': None, 'spk_encoder_type': None, 'content_encoder_type': None, 'content_encoder_ckpt': None}, 'model': {'save_dir': None, 'filter_channels': 768, 'finetune_from_model': {'discriminator': '/raid/lucasgris/free-svc/D-freevc-24.pth', 'generator': '/raid/lucasgris/free-svc/freevc-24.pth'}, 'hidden_channels': 192, 'inter_channels': 192, 'kernel_size': 3, 'n_heads': 2, 'n_layers_q': 3, 'n_layers': 6, 'p_dropout': 0.1, 'resblock_dilation_sizes': [[1, 3, 5], [1, 3, 5], [1, 3, 5]], 'resblock_kernel_sizes': [3, 7, 11], 'resblock': 1, 'c_dim': 768, 'upsample_initial_channel': 512, 'upsample_kernel_sizes': [16, 16, 4, 4], 'upsample_rates': [10, 8, 2, 2], 'use_spectral_norm': False, 'freeze_external_spk': True, 'device': 'cuda', 'gin_channels': None, 'post_content_encoder_type': 'vits-encoder-with-uv-emb', 'coarse_f0': True, 'cond_f0_on_flow': False, 'use_spk_emb': True, 'spk_encoder_type': 'ECAPA2SpeakerEncoder16k', 'spk_encoder_ckpt': None, 'content_encoder_type': 'spin', 'content_encoder_config': 'models/spin/spin.yaml', 'content_encoder_ckpt': 'models/spin/spin.ckpt'}}
> Loading model...
> Speaker encoder model: ECAPA2SpeakerEncoder16k
> /usr/local/lib/python3.10/dist-packages/torch/nn/utils/weight_norm.py:28: UserWarning: torch.nn.utils.weight_norm is deprecated in favor of torch.nn.utils.parametrizations.weight_norm.
>   warnings.warn("torch.nn.utils.weight_norm is deprecated in favor of torch.nn.utils.parametrizations.weight_norm.")
> SpinModel(
>   (encoder): HuBERT(
>     (model): HubertModel(
>       (feature_extractor): ConvFeatureExtractionModel(
>         (conv_layers): ModuleList(
>           (0): Sequential(
>             (0): Conv1d(1, 512, kernel_size=(10,), stride=(5,), bias=False)
>             (1): Dropout(p=0.0, inplace=False)
>             (2): Fp32GroupNorm(512, 512, eps=1e-05, affine=True)
>             (3): GELU(approximate='none')
>           )
>           (1-4): 4 x Sequential(
>             (0): Conv1d(512, 512, kernel_size=(3,), stride=(2,), bias=False)
>             (1): Dropout(p=0.0, inplace=False)
>             (2): GELU(approximate='none')
>           )
>           (5-6): 2 x Sequential(
>             (0): Conv1d(512, 512, kernel_size=(2,), stride=(2,), bias=False)
>             (1): Dropout(p=0.0, inplace=False)
>             (2): GELU(approximate='none')
>           )
>         )
>       )
>       (post_extract_proj): Linear(in_features=512, out_features=768, bias=True)
>       (dropout_input): Dropout(p=0.1, inplace=False)
>       (dropout_features): Dropout(p=0.1, inplace=False)
>       (encoder): TransformerEncoder(
>         (pos_conv): Sequential(
>           (0): Conv1d(768, 768, kernel_size=(128,), stride=(1,), padding=(64,), groups=16)
>           (1): SamePad()
>           (2): GELU(approximate='none')
>         )
>         (layers): ModuleList(
>           (0-11): 12 x TransformerSentenceEncoderLayer(
>             (self_attn): MultiheadAttention(
>               (dropout_module): FairseqDropout()
>               (k_proj): Linear(in_features=768, out_features=768, bias=True)
>               (v_proj): Linear(in_features=768, out_features=768, bias=True)
>               (q_proj): Linear(in_features=768, out_features=768, bias=True)
>               (out_proj): Linear(in_features=768, out_features=768, bias=True)
>             )
>             (dropout1): Dropout(p=0.1, inplace=False)
>             (dropout2): Dropout(p=0.0, inplace=False)
>             (dropout3): Dropout(p=0.1, inplace=False)
>             (self_attn_layer_norm): LayerNorm((768,), eps=1e-05, elementwise_affine=True)
>             (fc1): Linear(in_features=768, out_features=3072, bias=True)
>             (fc2): Linear(in_features=3072, out_features=768, bias=True)
>             (final_layer_norm): LayerNorm((768,), eps=1e-05, elementwise_affine=True)
>           )
>         )
>         (layer_norm): LayerNorm((768,), eps=1e-05, elementwise_affine=True)
>       )
>       (layer_norm): LayerNorm((512,), eps=1e-05, elementwise_affine=True)
>       (final_proj): None
>     )
>   )
>   (pred_head): DNN(
>     (layers): ModuleList(
>       (0): Linear(in_features=768, out_features=256, bias=True)
>     )
>     (dropout): Dropout(p=0, inplace=False)
>     (acts): ModuleList()
>   )
>   (loss_module): SwavVQDisentangle(
>     (codebook): Linear(in_features=256, out_features=2048, bias=False)
>   )
> )
> Loading checkpoint...
> Synthesizing...
> Original audio: 184920 Output audio: 184960
> Save path: gen-samples/2_src-2_ref_tgt1.wav

Any help and support would be appreciated,
Thanks!
